src/test/java/no/unit/nva/handlers/UpdateUserHandlerTest.java

```diff
@@ -1,65 +1,189 @@
+        + "an existing user id, and the body contains a valid UserDto and path id is the same as the body id ")
     @Test
     public void processInputReturnsUpdatedUserWhenPathAndBodyContainTheSameUserIdAndTheIdExistsAndBodyIsValid()
-        throws ApiGatewayException {
-        UserDto existingUser = storeInitialUserDirectlyToDatabase(createSampleUser());
+        throws ApiGatewayException, IOException {
 
-        UpdateUserHandler updateUserHandler = new UpdateUserHandler(envWithTableName, databaseService);
+        UserDto existingUser = storeUserInDatabase(createSampleUser());
+        UserDto userUpdate = createUserUpdate(existingUser);
+
+        GatewayResponse<Void> gatewayResponse = sendUpdateRequest(userUpdate.getUsername(), userUpdate);
+
+        Map<String, String> responseHeaders = gatewayResponse.getHeaders();
+
+        assertThat(responseHeaders, hasKey(LOCATION_HEADER));
+
+        String expectedLocationPath = UpdateUserHandler.USERS_RELATIVE_PATH + userUpdate.getUsername();
+        assertThat(responseHeaders.get(LOCATION_HEADER), is(equalTo(expectedLocationPath)));
+    }
+
+    @DisplayName("handleRequest() returns 202 (Accepted)  when path contains "
+        + "an existing user id, and the body contains a valid UserDto and path id is the same as the body id ")
+    @Test
+    public void processInputReturnsAcceptedWhenPathAndBodyContainTheSameUserIdAndTheIdExistsAndBodyIsValid()
+        throws ApiGatewayException, IOException {
+
+        UserDto existingUser = storeUserInDatabase(createSampleUser());
+        UserDto userUpdate = createUserUpdate(existingUser);
 
-        UserDto userUpdate = updateUser(existingUser);
-        RequestInfo requestInfo = createRequestInfoForQuery(existingUser);
-        UserDto actualUser = updateUserHandler.processInput(userUpdate, requestInfo, context);
+        GatewayResponse<Void> gatewayResponse = sendUpdateRequest(userUpdate.getUsername(), userUpdate);
 
-        assertThat(actualUser, is(equalTo(userUpdate)));
+        assertThat(gatewayResponse.getStatusCode(), is(equalTo(HttpStatus.SC_ACCEPTED)));
     }
 
-    private RequestInfo createRequestInfoForQuery(UserDto existingUser) {
-        RequestInfo requestInfo = new RequestInfo();
-        requestInfo.setPathParameters(
-            Collections.singletonMap(UpdateUserHandler.USER_ID_FIELD_NAME, existingUser.getUsername()));
-        return requestInfo;
+    @DisplayName("handleRequest() returns BadRequest  when path contains an Id that is different from the Id"
+        + "of the input Object")
+    @Test
+    public void processInputReturnsBadRequestWhenPathContainsIdDifferentFromIdOfInputObject()
+        throws ApiGatewayException, IOException {
+
+        UserDto existingUser = storeUserInDatabase(createSampleUser());
+        UserDto anotherExistingUser = createSampleUser().copy().withUsername(SOME_OTHER_USERNAME).build();
+        storeUserInDatabase(anotherExistingUser);
```

Comment:
Do you need to store the user in the DB? I don't think that this step should be necessary, just get the username